app/src/main/java/woowacourse/movie/view/home/adapter/MovieCatalogAdapter.kt

```diff
@@ -0,0 +1,62 @@
+package woowacourse.movie.view.home.adapter
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.RecyclerView
+import androidx.recyclerview.widget.RecyclerView.ViewHolder
+import woowacourse.movie.R
+import woowacourse.movie.model.advertisement.Advertisement
+import woowacourse.movie.model.movie.Movie
+import woowacourse.movie.view.home.adapter.viewholder.AdvertisementViewHolder
+import woowacourse.movie.view.home.adapter.viewholder.MovieViewHolder
+
+typealias OnClickReservationButton = (Movie) -> Unit
+
+class MovieCatalogAdapter(
+    private val movies: List<Movie>,
+    private val advertisements: List<Advertisement>,
+    private val movie: OnClickReservationButton,
+) : RecyclerView.Adapter<ViewHolder>() {
+    private val movieViewType = CatalogViewType.MOVIE.viewType
+    private val advertisementViewType = CatalogViewType.ADVERTISEMENT.viewType
+
+    override fun onCreateViewHolder(
+        parent: ViewGroup,
+        viewType: Int,
+    ): ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        return if (viewType == movieViewType) {
+            val view = inflater.inflate(R.layout.item_movie_catalog, parent, false)
+            MovieViewHolder(view)
+        } else {
+            val view = inflater.inflate(R.layout.item_advertisement, parent, false)
+            AdvertisementViewHolder(view)
+        }
+    }
+
+    override fun onBindViewHolder(
+        holder: ViewHolder,
+        position: Int,
+    ) {
+        when (getItemViewType(position)) {
+            movieViewType -> {
+                val item = movies[position]
+                (holder as MovieViewHolder).bind(item, movie)
+            }
+            advertisementViewType -> {
+                val item = advertisements[position - CatalogViewType.ADVERTISEMENT.position]
+                (holder as AdvertisementViewHolder).bind(item)
+            }
+        }
```

Comment:
```suggestion
        when (CatalogViewType.from(getItemViewType(position))) {
            CatalogViewType.ADVERTISEMENT -> {
                val item = movies[position]
                (holder as MovieViewHolder).bind(item, movie)
            }
            CatalogViewType.MOVIE -> {
                val item = advertisements[position - CatalogViewType.ADVERTISEMENT.position]
                (holder as AdvertisementViewHolder).bind(item)
            }
        }
        
```

```kotlin
// CatalogViewType.kt

companion object {
    fun from(viewType: Int): CatalogViewType {
        return entries.first { it.viewType == viewType }
    }
}
```

이런 식으로 하는 건 어떤지 제안해봅니다. 
enum 이 추가되었을때 빨간줄이 떠서 유지보수가 편합니다:)

Reply:
이런 방법이 있군요!! 확실히 enum 이 추가되었을 때 빨간줄이 떠서 유지보수가 편하겠네요 ㅎㅎ 감사합니다!
위 방식으로 수정해보았습니다! [f8c1380](https://github.com/woowacourse/android-movie-ticket/pull/80/commits/f8c1380ee56a597ee0f643fcd974e9a3c708d08b)

Reply:
지금은 ViewType을 관리하는 enum에서 viewType이라는 별도의 속성을 추가해서 관리하는 데, 이를 없애고 enum의 ordinal를 활용하는 것에 대해 파이로는 어떻게 생각하시는 지 여쭤봐도 될까요? 제가 처음에 ordinal를 활용하지 않고 직접 viewType 속성을 만든 이유는 ordinal이 명시적이지 않다고 생각해 유지보수가 힘들 것 같아서 사용하지 않았습니다!